lib/kernel.ts

```diff
@@ -155,64 +185,44 @@ const getSelected = (schema: JsonSchema): { links: any; properties: any } => {
 	};
 };
 
-const rectifySelected = (
-	selected: any,
-	filteredSelected: { [x: string]: any },
-) => {
-	for (const [key, value] of Object.entries(selected)) {
-		if (key in filteredSelected) {
-			rectifySelected(value, filteredSelected[key]);
+// TODO: rename with something more descriptive.
+const rectifySelectObject = (
+	selectObject: SelectObject,
+	authorizedSelectObject: SelectObject,
+): SelectObject => {
+	const result = { ...selectObject };
```

Comment:
It doesn't need to be cloned

Reply:
dealing with ownership in most dynamic languages is not great :/ but in this case it's not necessary to preserve the original one